generators/typescript/asIs/tests/mock-server/mockEndpointBuilder.ts

```diff
@@ -0,0 +1,209 @@
+import { DefaultBodyType, HttpHandler, HttpResponse, HttpResponseResolver, http } from "msw";
+
+import { toJson } from "../../src/core/json";
+import { withHeaders } from "./withHeaders";
+import { withJson } from "./withJson";
+
+type HttpMethod = "all" | "get" | "post" | "put" | "delete" | "patch" | "options" | "head";
+
+interface MethodStage {
+    baseUrl(baseUrl: string): MethodStage;
+    all(path: string): RequestHeadersStage;
+    get(path: string): RequestHeadersStage;
+    post(path: string): RequestHeadersStage;
+    put(path: string): RequestHeadersStage;
+    delete(path: string): RequestHeadersStage;
+    patch(path: string): RequestHeadersStage;
+    options(path: string): RequestHeadersStage;
+    head(path: string): RequestHeadersStage;
+}
+
+interface RequestHeadersStage extends RequestBodyStage, ResponseStage {
+    header(name: string, value: string): RequestHeadersStage;
+    headers(headers: Record<string, string>): RequestBodyStage;
+}
+
+interface RequestBodyStage extends ResponseStage {
+    jsonBody(body: unknown): ResponseStage;
+}
+
+interface ResponseStage {
+    respondWith(): ResponseStatusStage;
+}
+interface ResponseStatusStage {
+    statusCode(statusCode: number): ResponseHeaderStage;
+}
+
+interface ResponseHeaderStage extends ResponseBodyStage, BuildStage {
+    header(name: string, value: string): ResponseHeaderStage;
+    headers(headers: Record<string, string>): ResponseHeaderStage;
+}
+
+interface ResponseBodyStage {
+    jsonBody(body: unknown): BuildStage;
+}
+
+interface BuildStage {
+    build(): HttpHandler;
+}
+
+export interface HttpHandlerBuilderOptions {
+    onBuild?: (handler: HttpHandler) => void;
+    once?: boolean;
+}
+
+class RequestBuilder implements MethodStage, RequestHeadersStage, RequestBodyStage, ResponseStage {
+    private method: HttpMethod = "get";
+    #baseUrl: string = "";
+    private path: string = "/";
+    private readonly predicates: ((resolver: HttpResponseResolver) => HttpResponseResolver)[] = [];
+    private readonly handlerOptions?: HttpHandlerBuilderOptions;
+
+    constructor(options?: HttpHandlerBuilderOptions) {
+        this.handlerOptions = options;
+    }
+
+    baseUrl(baseUrl: string): MethodStage {
+        this.#baseUrl = baseUrl;
+        return this;
+    }
+
+    all(path: string): RequestHeadersStage {
```

Comment:
When is this required?

Reply:
We don't use it, I'm just mirroring the available methods on MSW, but we can reduce it to the ones we use ourselves.
However, if a SDK author (or us) wants to use this for handwritten tests, it could come in handy.